cdap-ui/app/directives/start-stop-button/start-stop-button.js

```diff
@@ -44,7 +44,12 @@ angular.module(PKG.name + '.commons')
           }
           dataSrc.request(requestObj)
             .then(function() {
-              $state.go($state.current, $state.params, {reload: true});
+              if ($state.params.runid && action === 'start') {
```

Comment:
Talked to @elfenheart. Changing this to use ```$state.is``` to check if we are in /runs or /runs/:runid states.